units/pod_termination.go

```diff
@@ -214,3 +206,39 @@ func (j *podTerminationJob) fixStrandedTasks(ctx context.Context) error {
 
 	return nil
 }
+
+func (j *podTerminationJob) fixStrandedRunningTask(ctx context.Context) error {
+	if j.pod.TaskRuntimeInfo.RunningTaskID == "" {
+		return nil
```

Comment:
A grip statement in this case may be good for monitoring

Reply:
This is the normal case since the pod would only be terminated while still running a task in a few edge cases. I think it's still worth tracking when we clear and reset stranded tasks though, so I added a log for cases where we clear and reset tasks that are stale or stranded.